Assets/Scripts/Transformation.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Transformation : MonoBehaviour {
+
+	[SerializeField] private float transDuration = 1.2f, humanFormDuration = 5f;
+
+	private SpriteRenderer mySpriteRend;
+
+	private TransformationStates transStates;
+
+	private Color humanColor = Color.white, wolfColor = Color.black, transformingColor = Color.grey;
+
+	private float transTimer, humanFormTimer;
+
+
+	private void Start() {
+		mySpriteRend = gameObject.GetComponent<SpriteRenderer>();
+
+		mySpriteRend.color = wolfColor;
+	}
+
+
+	private void Update() {
+		if (Input.GetKeyDown(KeyCode.R)) {
+			StartCoroutine(CoTransforming());
+		}
+	}
+
+
+	private IEnumerator CoTransforming() {
+		if (transStates == TransformationStates.Wolf) {
+
+			mySpriteRend.color = transformingColor;
+	
+			for (transTimer = transDuration; transTimer > 0 ; transTimer -= Time.deltaTime) {
+				if (Input.GetKeyUp(KeyCode.R)) {
+					mySpriteRend.color = wolfColor;
+					yield break;
+				}
+			yield return null;
+			}
+		}
+		else if (transStates == TransformationStates.Human) {
+
+			mySpriteRend.color = transformingColor;
+
+			for (transTimer = transDuration; transTimer > 0 ; transTimer -= Time.deltaTime) {
+				if (Input.GetKeyUp(KeyCode.R)) {
+					mySpriteRend.color = humanColor;
+					yield break;
+				}
+				yield return null;
+			}
+		}
```

Comment:
Duplicate code in if statement blocks. The only difference is what color you set so you only need to check when you set the color.
```suggestion
		mySpriteRend.color = transformingColor;

		for (transTimer = transDuration; transTimer > 0 ; transTimer -= Time.deltaTime) {
			if (Input.GetKeyUp(KeyCode.R)) {
				if (transStates == TransformationStates.Wolf)
					mySpriteRend.color = wolfColor;
				else if (transStates == TransformationStates.Human)
					mySpriteRend.color = humanColor;

				yield break;
			}
			yield return null;
		}
```